kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ProcessServiceIntegrationTest.java

```diff
@@ -428,6 +428,79 @@ public void testManipulateProcessVariable() throws Exception {
         }
 
     }
+    
+    @Test
+    public void testProcessVariablesDataTypes() throws Exception {
+        Long processInstanceId = processClient.startProcess(CONTAINER_ID, "variables.types");
+        assertNotNull(processInstanceId);
+        assertTrue(processInstanceId>0);
+        try {
+            
+            short age = 44;
+            processClient.setProcessVariable(CONTAINER_ID, processInstanceId, "age", age );
+            assertEquals( age, ((Short)processClient.getProcessInstanceVariable(CONTAINER_ID, processInstanceId, "age")).shortValue());
+            
+            boolean married = true;
+            processClient.setProcessVariable(CONTAINER_ID, processInstanceId, "married", married );
+            assertEquals( married, ((Boolean)processClient.getProcessInstanceVariable(CONTAINER_ID, processInstanceId, "married")).booleanValue());
```

Comment:
unnecessary unboxing. You could also use a different assert
```suggestion
            assertTrue((Boolean)processClient.getProcessInstanceVariable(CONTAINER_ID, processInstanceId, "married"));
```

Reply:
but here I want to compare with the value of married, that might be true or false, so I think equals should remain and I want to compare the boolean, so the unboxing is needed